source/collections/Queue.ooc

```diff
@@ -12,17 +12,20 @@ Queue: abstract class <T> {
 	dequeue: abstract func -> (T, Bool)
 	peek: abstract func -> (T, Bool)
 	//TODO: This works much better and will do for now
-	dequeue: abstract func ~out (result: T*) -> Bool
+	dequeue: abstract func ~out (result: T*) -> Bool // Also crashing the compiler
```

Comment:
The new methods works for both classes and covers but covers need a reserved value when using the thread safe queue since you can't ask if it is empty and then assume that no other thread came before you.